client/device/cloudftd/delete.go

```diff
@@ -36,7 +42,95 @@ func Delete(ctx context.Context, client http.Client, deleteInp DeleteInput) (*De
+	// 2.5.1 read FTD name from input uid
+	readFtdOutp, err := ReadByUid(ctx, client, NewReadByUidInput(deleteInp.Uid))
+	if err != nil {
+		return nil, err
+	}
+	// 2.5.2 read all device records
+	allDeviceRecords, err := fmcconfig.ReadAllDeviceRecords(ctx, client, fmcconfig.NewReadAllDeviceRecordsInput(fmcDomainUid, fmcReadRes.Host))
+	if err != nil {
+		return nil, err
+	}
+	var ftdRecordId string
+	// 2.5.3 check if FTD name is present in device records, logic: same name + both are FTDs = found
+	client.Logger.Printf("looking for existing FTD with id=%s and name=%s\n", deleteInp.Uid, fmcReadRes.Name)
+	for _, record := range allDeviceRecords.Items {
+		if record.Name != readFtdOutp.Name {
+			// different name, ignore
+			continue
+		}
+		// the allDeviceRecords only contains the name, so we need to make another call to retrieve the details of the device to check whether this is a FTD
+		// potentially we will be making a lot of network calls and cause this loop to run for long time if
+		// we have many device records with the same name, I suppose that rarely happens
+		deviceRecord, err := fmcconfig.ReadDeviceRecord(ctx, client, fmcconfig.NewReadDeviceRecordInput(fmcDomainUid, fmcReadRes.Host, record.Id))
+		if err != nil {
+			return nil, err
+		}
+		if strings.Contains(deviceRecord.Model, "Firepower Threat Defense") { // Question: is there a better way to check? Does this check cover all cases?
+			// found
+			ftdRecordId = record.Id
+		} else {
+			// not a FTD, just some other device with the same name, ignore
+		}
+	}
+
+	if ftdRecordId != "" {
+		// FTD record found in the cdFMC device records
+		// now use the FTD device record id and to read the device record, until the FTD deployment status is DEPLOYED
+
+		// 2.5.4 re-read FMC's FTD deployment status until it is DEPLOYED
+		err = retry.Do(
+			ctx,
+			func() (bool, error) {
+				ftdDeviceRecord, err := fmcconfig.ReadDeviceRecord(ctx, client, fmcconfig.NewReadDeviceRecordInput(fmcDomainUid, fmcReadRes.Host, ftdRecordId))
+				if err != nil {
+					return false, err
+				}
+				client.Logger.Printf("current deployment status=%s\n", ftdDeviceRecord.DeploymentStatus)
+				if ftdDeviceRecord.DeploymentStatus == "DEPLOYED" { // no idea what this deployment status could be, so it is a string
+					return true, nil
+					// TODO: check for error here: like if ftdDeviceRecord.DeploymentStatus == "DEPLOY_ERROR" {, not sure what is the error deployment status so I did not do it for now
+				} else {
+					return false, nil
+				}
+			},
+			retry.NewOptionsBuilder().
+				Message("Waiting for FTD deployment to finish...").
+				Timeout(15*time.Minute). // usually 5-10 minutes
+				Delay(3*time.Second).
+				Logger(client.Logger).
+				Retries(-1).
+				EarlyExitOnError(true).
+				Build(),
+		)
+		if err != nil {
+			// error during retry, maybe the deployment failed, we fail the delete as well
+			return nil, err
+		}
+
+	} else {
```

Comment:
Do we need this empty else?

Reply:
No, it is just to put the comment, will remove the `else` 👍